server/src/main/java/org/elasticsearch/cluster/metadata/IndexAbstraction.java

```diff
@@ -238,7 +238,7 @@ public Alias(AliasMetadata aliasMetadata, List<IndexMetadata> indices) {
 
             List<IndexMetadata> writeIndices = indices.stream()
                 .filter(idxMeta -> Boolean.TRUE.equals(idxMeta.getAliases().get(aliasName).writeIndex()))
-                .collect(Collectors.toList());
+                .collect(Collectors.toCollection(ArrayList::new));
```

Comment:
Is it equivalent to `Collectors.toList()`? Am I missing something?

Reply:
It is, but with the important distinction that this respects the JDKs compatibility guarantees.
The docs to `Collectors.toList` state:

> returns a {@code Collector} that accumulates the input elements into a
     new {@code List}. There are no guarantees on the type, mutability, serializability, or thread-safety of the {@code List} returned;

So using this runs some risks of JDK changes breaking things unexpectedly and also makes the code behave in a surprising way (I certainly was surprised to find the spots I had to adjust here).

Reply:
>we rely on the not-guaranteed behavior of Collectors.toList() producing a mutable List

:+1: